### ISSUE TITLE
How to get the contained type T of a synchronized_value<T>

### ISSUE THREAD

**Issue by davidhunter22** [jrgfogh/synchronized_value#5]
Given the following
```
    using X = synchronized_value<std::string>;
```
if I am given type X how do I retrieve the contained type, in this case std::string;
There seem to be no
```
    using value_type = T;
```
in the synchronized_value class.

BTW thanks for working on this a very needed feature.

**Comment by jrgfogh** [comment]
This definitely belongs in `synchronized_value`.

I originally wrote it based on the proposal for the standard, which does not include `value_type`. There are so many things I want to improve that I have decided to start from scratch.

If you can't wait, you're welcome to open a pull request if you want it added to the current version.

**Comment by jrgfogh** [comment]
Solved in PR #7.
